framework/src/loops/ComputeMortarFunctor.C

```diff
@@ -38,7 +38,11 @@ ComputeMortarFunctor::ComputeMortarFunctor(
 {
   // Construct the mortar constraints we will later loop over
   for (auto mc : mortar_constraints)
+  {
     _mortar_constraints.push_back(mc.get());
+    // Whether to use dual mortar. Same for all motrat constraints
+    _use_dual = mc->use_dual();
```

Comment:
Hmm why do we have to do this? Now that I'm thinking about it, I can actually fairly easily imagine non-dual and dual LM calculations within the same simulation *and* with different orders for the LMs. Any simulation with thermal and mechanical contact. Thermal contact does not have any saddle point issues and hence does not have a clear reason for using dual; moreover, you get optimal convergence in the primal variable if you use an order for the thermal LM of (primal_order - 1). I think that we need to figure out a way *not* to do what you're doing here. (Also `_use_dual` in `ComputeMortarFunctor` would just end up corresponding to whatever the last mortar constraint's `_use_dual` value was based on what you're doing here)

Reply:
> Hmm why do we have to do this? Now that I'm thinking about it, I can actually fairly easily imagine non-dual and dual LM calculations within the same simulation _and_ with different orders for the LMs. Any simulation with thermal and mechanical contact. Thermal contact does not have any saddle point issues and hence does not have a clear reason for using dual; moreover, you get optimal convergence in the primal variable if you use an order for the thermal LM of (primal_order - 1). I think that we need to figure out a way _not_ to do what you're doing here. (Also `_use_dual` in `ComputeMortarFunctor` would just end up corresponding to whatever the last mortar constraint's `_use_dual` value was based on what you're doing here)

The purpose was to pass in the `use_dual` input parameter in order to initialize the lowerD elements accordingly. I have not met cases where different orders for the LMs exist (and not sure if that would necessarily be a common practice). So I just left a comment there and assumed same`use_dual` value for all mortar constraints.

Not sure how to change it at this moment, but I probably can address issue while dealing with the ones you pointed out later in this PR.

Reply:
> I have not met cases where different orders for the LMs exist for one simulation (and not sure if that would necessarily be a common practice).

See https://github.com/idaholab/moose/issues/13080#issuecomment-475752613. If using first order displacements, you will want to use a non-dual CONSTANT MONOMIAL for the thermal contact Lagrange multiplier. If using second order displacements, you would want to use a LAGRANGE FIRST for the thermal contact LM. Based on what you told me, you will be wanting to use an order equal to the displacement order for the mechanical contact LM. In any realistic BISON simulation you will have both thermal and mechanical contact and so you will have mixing of dual and non-dual and multiple LM orders.